### PR TITLE
fix(FeedSourceController): delete feed zip when feed becomes private

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [x] The description lists all applicable issues this PR seeks to resolve
- [x] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

Currently when setting a feed source to private the feed source is removed from the generated `index.html`, but the file is still present so anyone who has that link will still get access to that previous feed. This PR repairs this, deleting the zip file from S3 when it is set to private.

### CODE REVIEW

File: src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java [hunk 1]
```diff
@@ -201,6 +203,12 @@ private static FeedSource updateFeedSource(Request req, Response res) throws IOE
         }
         Persistence.feedSources.replace(feedSourceId, updatedFeedSource);
 
+        // If feed just changed from public to private, delete feed from public repo if it's present there
```
Comment (br648) [inline src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java]: Nitpicking, full stop.

File: src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java [hunk 2]
```diff
@@ -201,6 +203,12 @@ private static FeedSource updateFeedSource(Request req, Response res) throws IOE
...
+        if(formerFeedSource.isPublic && !updatedFeedSource.isPublic) {
```
Comment (br648) [inline src/main/java/com/conveyal/datatools/manager/controllers/api/FeedSourceController.java]: Nitpicking, space between the "if" and open bracket.

**Review by br648 — APPROVED** [review]
LGTM (At least from the nit picking you know I've actually reviewed it!)

**Comment by miles-grant-ibigroup** [comment]
Thanks for the review!

**Review by philip-cline — APPROVED** [review]
Stupendous & beautiful. Thank you for the fix.

**Pull request merged**